app/controllers/events_controller.rb

```diff
@@ -0,0 +1,75 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_project_member
+  before_action :authorize_event_author, only: [:update, :destroy]
+
+  def index
+    @project = Project.find(params[:project_id])
+  end
+
+  def show
+  end
+
+  def new
+    @event = Project.find(params[:project_id]).events.build
+  end
+
+  def edit
+  end
+
+  def create
+    @event = Project.find(params[:project_id]).events.build(event_params)
```

Comment:
you can extract @project = Project.find(params[:project_id] to some private method set_project since you use it a few times in your controller